cmd/bake/bake.go

```diff
@@ -67,19 +70,74 @@ type bakePostRequest struct {
 	Wait bool   `json:"wait,omitempty"`
 }
 
+type repos struct {
+	URLs []string `yaml:"repos"`
+}
+
+type bakeStatus struct {
+	URL string
+	Err error
+}
+
 func run(opts *Options) error {
-	bodyPostReq := &bakePostRequest{
-		URL:  opts.URL,
-		Wait: opts.Wait,
+	var repos repos
+	uniqueURLs := make(map[string]int)
+
+	if opts.FilePath != "" {
+		configFile, err := os.ReadFile(opts.FilePath)
+
+		if err != nil {
+			return err
+		}
+
+		err = yaml.Unmarshal(configFile, &repos)
+
+		if err != nil {
+			return err
+		}
+
+		for _, url := range repos.URLs {
+			uniqueURLs[url] = 0
+		}
 	}
 
+	// make sure there are no duplicated queries to the same URL
+	for _, url := range opts.URLs {
+		if _, exists := uniqueURLs[url]; !exists {
+			uniqueURLs[url] = 0
+		}
+	}
+
+	var batchStatus []bakeStatus
+
+	for url := range uniqueURLs {
+		bodyPostReq := bakePostRequest{
+			URL:  url,
+			Wait: opts.Wait,
+		}
+
+		err := bakeRepo(bodyPostReq, opts.Endpoint)
+		batchStatus = append(batchStatus, bakeStatus{URL: url, Err: err})
+	}
```

Comment:
A future optimization would be to put this all behind a `go func` so that we can batch all the requests at once. The server is able to handle many requests and this would be a good optimization instead of processing each one by one.

But this is fine for now, we can make an issue for that after the fact.